### DISCUSSION TITLE
Unixtime

### DISCUSSION THREAD

**Discussion by vovacha** [gamedaoco/gamedao-protocol#82] [General]
We have `created` Campaign's field as a UnixTime type. And since it's only one pallet where we have it, I'm questioning if we need it or not.

I've briefly reviewed Substrate and Acala repos:
- FRAME, only one pallet - [Staking uses UnixTime to compute payout for the era](https://github.com/paritytech/substrate/blob/master/frame/staking/src/pallet/impls.rs#L377-L381).
- As for the Acala, one module - [cdp-engine uses it to get time intervals in seconds](https://github.com/AcalaNetwork/Acala/blob/master/modules/cdp-engine/src/lib.rs#L596).

It looks like it's not so extensively used, but let's discuss why we need it and the pros/cons.

@2075:
> i would not agree that it is not a common practice, most timestamps in tech are based on unix timestamps. but i agree that we could store the start block, but we will possibly not be able to derive a reliable timestamp for the campaign anymore. (also true for all other transactions which could end in an audit log or so, but maybe also reasonable to store events with there timestamp separately, so this could be removed eventually)

**Comment by 2075** [comment]
My take on this is roughly this:

- wherever you have absolute time dependent decisions and calculations for tx, keep unix timestamp
- whenever granularity and stability of blocktime is granular enough, use blocktime.

**Reply by vovacha** [reply]
And talking about the Campaign creation time: is block time granular enough, or do we need a timestamp? What's the use case of `created` field? Showing it on UI side? No backend logic connected to it so far.
